Bugzilla/Mailer.pm

```diff
@@ -120,37 +117,17 @@ sub MessageToMTA {
 
   my $from = $email->header('From');
 
-  my ($hostname, @args);
-  my $mailer_class = $method;
-  if ($method eq "Sendmail") {
-    $mailer_class = 'Bugzilla::Send::Sendmail';
-    if (ON_WINDOWS) {
-      $Email::Send::Sendmail::SENDMAIL = SENDMAIL_EXE;
-    }
-    push @args, "-i";
-
-    # We want to make sure that we pass *only* an email address.
-    if ($from) {
-      my ($email_obj) = Email::Address->parse($from);
-      if ($email_obj) {
-        my $from_email = $email_obj->address;
-        push(@args, "-f$from_email") if $from_email;
-      }
-    }
-  }
-  else {
-    # Sendmail will automatically append our hostname to the From
-    # address, but other mailers won't.
-    my $urlbase = Bugzilla->localconfig->urlbase;
-    $urlbase =~ m|//([^:/]+)[:/]?|;
-    $hostname = $1;
-    $from .= "\@$hostname" if $from !~ /@/;
-    $email->header_set('From', $from);
-
-    # Sendmail adds a Date: header also, but others may not.
-    if (!defined $email->header('Date')) {
-      $email->header_set('Date', time2str("%a, %d %b %Y %T %z", time()));
-    }
+  # Sendmail will automatically append our hostname to the From
+  # address, but other mailers won't.
+  my $urlbase = Bugzilla->localconfig->urlbase;
+  $urlbase =~ m|//([^:/]+)[:/]?|;
```

Comment:
Is this variable used anywhere?

Reply:
yes and no. We save `Bugzilla->localconfig->urlbase` value into a variable so we can perform a proper regex on the value. Then in the next line, you will see `$urlbase` with a regex match performed against it. And then next, `$hostname = $1;` is using the matched content from inside the regex parenthesis from the previous regex. So `$urlbase` is used but only briefly.